### PR TITLE
fix test when day is past 27

### PR DESCRIPTION
**Pull request by nilsnolde**
when the day is past 27 (could be 28 really), it would create a month string of e.g. `5` instead of `05`, so the test would fail.

### CODE REVIEW

File: test/gurka/test_gtfs.cc [hunk 1]
```diff
@@ -862,6 +862,7 @@ TEST(GtfsExample, route_trip1) {
   // wrap around the next month if it's getting critical
   if (tmrw_int > 27) {
     auto new_month = std::to_string(std::stoul(std::string(&req_time[5], &req_time[7])) + 1);
+    new_month = std::string(2 - std::min(2UL, new_month.length()), '0') + new_month;
```
Comment (nilsnolde) [inline test/gurka/test_gtfs.cc:865]: fill the month string with a leading zero

File: test/gurka/test_gtfs.cc [hunk 2]
```diff
@@ -862,6 +862,7 @@ TEST(GtfsExample, route_trip1) {
   // wrap around the next month if it's getting critical
   if (tmrw_int > 27) {
     auto new_month = std::to_string(std::stoul(std::string(&req_time[5], &req_time[7])) + 1);
```
Comment (kevinkreiser) [inline test/gurka/test_gtfs.cc:864]: and when its december? shouldnt we let the date time library handle this. it has support for addition and subtraction but requires a little bit more work and temporary values of course..

**Review by kevinkreiser — APPROVED** [review]
this will work for now, but when its december its going to  be very hard to build the test in the 13th month of the year :smile:

File: test/gurka/test_gtfs.cc (hunk 2 above)
Comment (nilsnolde) [reply test/gurka/test_gtfs.cc:864]: argh.. right.. I'll write an issue about it. I'm sure I'll get to it before 2024;)

**Comment by nilsnolde** [comment]
feel free to force push (no need for changelog IMO). I left an issue to fix this within the next 7 months:)

**Pull request merged**